src/Test/L0/Util/RepositoryUtilL0.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using Agent.Sdk;
+using Microsoft.TeamFoundation.DistributedTask.Pipelines;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.Util
+{
+    public sealed class RepositoryUtilL0
+    {
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Common")]
+        public void HasMultipleCheckouts_should_not_throw()
+        {
+            using (TestHostContext hc = new TestHostContext(this))
+            {
+                Tracing trace = hc.GetTrace();
+
+                Assert.Equal(false, RepositoryUtil.HasMultipleCheckouts(null));
+                var dict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                Assert.Equal(false, RepositoryUtil.HasMultipleCheckouts(dict));
+                dict.Add("x", "y");
+                Assert.Equal(false, RepositoryUtil.HasMultipleCheckouts(dict));
+                dict.Add(WellKnownJobSettings.HasMultipleCheckouts, "burger");
+                Assert.Equal(false, RepositoryUtil.HasMultipleCheckouts(dict));
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Common")]
+        public void HasMultipleCheckouts_should_return_true_when_set_correctly()
+        {
+            using (TestHostContext hc = new TestHostContext(this))
+            {
+                Tracing trace = hc.GetTrace();
+
+                Assert.Equal(false, RepositoryUtil.HasMultipleCheckouts(null));
+                var dict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                dict[WellKnownJobSettings.HasMultipleCheckouts] = "true";
+                Assert.Equal(true, RepositoryUtil.HasMultipleCheckouts(dict));
+                dict[WellKnownJobSettings.HasMultipleCheckouts] = "TRUE";
+                Assert.Equal(true, RepositoryUtil.HasMultipleCheckouts(dict));
+                dict[WellKnownJobSettings.HasMultipleCheckouts] = "True";
+                Assert.Equal(true, RepositoryUtil.HasMultipleCheckouts(dict));
```

Comment:
what about false-y strings?

Reply:
Fair point. I can add some tests to make sure we get what is expected.